kura/org.eclipse.kura.asset.provider/src/main/java/org/eclipse/kura/asset/provider/BaseAsset.java

```diff
@@ -555,9 +580,14 @@ public void onDriverEvent(final DriverEvent event) {
         final Set<String> channelPrefixes = CollectionUtil.newHashSet();
         for (final Map.Entry<Long, Channel> entry : channels.entrySet()) {
             final Long key = entry.getKey();
-            final String prefix = key + CHANNEL_PROPERTY_POSTFIX.value() + CHANNEL_PROPERTY_PREFIX.value()
-                    + CHANNEL_PROPERTY_POSTFIX.value();
-            channelPrefixes.add(prefix);
+
+            final StringBuilder channelPrefix = new StringBuilder();
```

Comment:
Not really. The string concatenation, if performed, is done at different steps (at compile time or using JIT). The current Java8 for ARM is provided without JIT, so in our target hardware, it is fully interpreted and the concatenation optimizations are not performed.
Finally, having a look [at this research](http://www.pellegrino.link/2015/08/22/string-concatenation-with-java-8.html), using explicitly StringBuilder is still the best option.